packages/tools/kadena-cli/src/account/commands/accountFund.ts

```diff
@@ -1,73 +1,147 @@
-import type { ChainId } from '@kadena/client';
-import { createSignWithKeypair } from '@kadena/client';
-import { transfer } from '@kadena/client-utils/coin';
+import { describeModule } from '@kadena/client-utils/built-in';
 import chalk from 'chalk';
-import type { Command } from 'commander';
+import { Option } from 'commander';
 import debug from 'debug';
-import { FAUCET_CONSTANTS } from '../../constants/faucet.js';
-import { createCommand } from '../../utils/createCommand.js';
+import ora from 'ora';
+import { z } from 'zod';
+import { FAUCET_MODULE_NAME } from '../../constants/devnets.js';
+import deployDevNetFaucet from '../../devnet/faucet/deploy/index.js';
+import { networkIsAlive } from '../../devnet/utils/network.js';
+import { loadNetworkConfig } from '../../networks/utils/networkHelpers.js';
+import { actionAskForDeployDevnet } from '../../prompts/genericActionPrompts.js';
+import { networkSelectOnlyPrompt } from '../../prompts/network.js';
+import { assertCommandError } from '../../utils/command.util.js';
+import { createCommandFlexible } from '../../utils/createCommandFlexible.js';
+import { createOption } from '../../utils/createOption.js';
 import { globalOptions } from '../../utils/globalOptions.js';
+import { fund } from '../utils/fund.js';
 
-export const fundCommand: (
-  program: Command,
-  version: string,
-) => void = createCommand(
+const networkSelect = createOption({
+  key: 'network' as const,
+  prompt: networkSelectOnlyPrompt,
+  validation: z.string(),
+  option: new Option(
+    '-n, --network <network>',
+    'Kadena network (e.g. "mainnet")',
+  ),
+  expand: async (network: string) => {
+    try {
+      const networkConfig = loadNetworkConfig(network);
+      return networkConfig;
+    } catch (e) {
+      throw new Error(
+        `No network configuration found for "${network}". Please create a "${network}" network.`,
+      );
+    }
+  },
+});
+
+const deployDevnet = createOption({
+  key: 'deployDevnet',
+  validation: z.boolean(),
+  prompt: actionAskForDeployDevnet,
+  option: new Option('-d, --deploy-devnet', 'Deploy devnet if not available.'),
+});
+
+/* bin/kadena-cli.js account fund --account="dev.yaml" --amount="20" --network="devnet" --chain-id="0" */
+
+export const createFundCommand = createCommandFlexible(
   'fund',
-  'fund account',
+  'fund an existing/new account',
   [
-    globalOptions.accountName(),
+    globalOptions.accountSelect(),
     globalOptions.amount(),
-    globalOptions.fungible(),
-    globalOptions.network(),
+    networkSelect(),
     globalOptions.chainId(),
+    deployDevnet(),
```

Comment:
Devnet should be disabled, in this pr, since we are not shipping devnet support with the first release